include/deal.II/base/symmetric_tensor.h

```diff
@@ -2096,6 +2099,42 @@ SymmetricTensor<rank,dim,Number>::serialize(Archive &ar, const unsigned int)
 
 
 /**
+ * Add two symmetric tensors. If possible, you should use <tt>operator +=</tt>
+ * instead since this does not need the creation of a temporary.
+ *
+ * @relates SymmetricTensor
+ */
+template <int rank, int dim, typename Number, typename OtherNumber>
+inline
+SymmetricTensor<rank, dim, typename ProductType<Number, OtherNumber>::type>
+operator + (const SymmetricTensor<rank, dim, Number>      &left,
```

Comment:
So I've been looking at this all day, and haven't had much luck at implementing a `<Sum/Addition/Subtraction>Type` that is compatible with `Sacado::Fad::DFad`. The math-ops create expressions of type `Sacado::Fad::Expr<Op>`, which are not trivially convertible from a `Sacado::Fad::DFad`, meaning that some of the initialisations implemented in the `Tensor` and `SymmetricTensor` classes do not work. However, it is possible to construct a `Sacado::Fad::DFad` directly from any `Sacado::Fad::Expr<Op>`, which is why the current implementation works (the result of any `ProductType` invoving a `Sacado::Fad::DFad` is the Sacado number).

If nobody objects, I'd like to leave this as is for now but I will try to fix it at a later stage. I agree that our current implementation is wrong, but it works fine at the moment (presumably at the expense of some performance).